File: gluecodium/src/test/resources/smoke/listeners/output/android/com/example/smoke/InternalListenerImpl.java
/*
 *

 */
package com.example.smoke;
import com.example.NativeBase;
class InternalListenerImpl extends NativeBase implements InternalListener {
    protected InternalListenerImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    public native void onEvent();
}

File: gluecodium/src/test/resources/smoke/listeners/output/android-kotlin/com/example/smoke/InternalListenerImpl.kt
/*

 *
 */

@file:JvmName("InternalListenerImplExtensions")


package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
internal class InternalListenerImpl : NativeBase, InternalListener {
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
